File: functions/database/database-player.js
function DatabasePlayer(database, document)
{
	this.byGameRef = function(gameRef)
	{
		return new Promise((resolve, reject) => 
		{
			gameRef.collection('players').get()
			.then(docList =>
			{
				const accountRefs = docList.docs.map(doc => doc.data().account)

				database.account.listByRef(accountRefs)
				.then(accountDocs =>
				{
					const playerDocs = []

					for (var i = 0; i < docList.docs.length; i++)
					{
						playerDocs.push(new document.player(docList.docs[i], accountDocs[i]))
					}

					resolve(playerDocs)
				})
			})
		})
	}
}

module.exports = DatabasePlayer

File: src/main/kotlin/com/mauriciotogneri/momowars/database/DatabasePlayer.kt
package com.mauriciotogneri.momowars.database

import com.mauriciotogneri.momowars.document.DocumentAccount
import com.mauriciotogneri.momowars.document.DocumentPlayer
import com.mauriciotogneri.momowars.firebase.DocumentReference
import com.mauriciotogneri.momowars.utils.await

class DatabasePlayer(private val database: Database)
{
    suspend fun byGameRef(gameRef: DocumentReference): List<DocumentPlayer>
    {
        val snapshot = gameRef.collection("players").get().await()

        return snapshot.docs.map {
            DocumentPlayer(it, documentAccount(it.data().account))
        }
    }

    private suspend fun documentAccount(ref: DocumentReference): DocumentAccount = DocumentAccount(ref.get().await())
}